pkg/git/main.go

```diff
@@ -89,10 +101,17 @@ func GetGitPlan(plan string, localShuttleDirectoryPath string, uii ui.UI, skipGi
 				uii.Verboseln("Skipping git plan pulling")
 				return planPath
 			}
-
-			uii.Infoln("Pulling latest plan changes")
+			gitCmd("fetch origin", planPath, uii)
+			gitCmd(fmt.Sprintf("checkout %s", parsedGitPlan.head), planPath, uii)
+			status := getStatus(planPath)
+			if !status.isDetached {
+				uii.Infoln("Pulling latest plan changes on %v", parsedGitPlan.head)
+				gitCmd("pull origin", planPath, uii)
+				status = getStatus(planPath)
+			} else {
+				uii.EmphasizeInfoln("Skipping plan pull because its running on detached head")
```

Comment:
What does this mean? When can I get into this case? If I specify a specific commit SHA or?

Reply:
When you checkout a SHA or tag git will get into detached mode. This is what this means.
Not sure how to explain it further without getting into a more complex sentence.

Reply:
It's fine. The question was out of curiosity. :)